server/controller/src/main/java/ai/starwhale/mlops/api/DataStoreController.java

```diff
@@ -165,14 +165,32 @@ public ResponseEntity<ResponseMessage<RecordListVo>> scanTable(ScanTableRequest
     @Override
     public void queryAndExport(QueryTableRequest request, HttpServletResponse httpResponse) {
         try {
-            RecordList recordList = queryRecordList(request);
-            byte[] bytes = recordsExporter.asBytes(recordList);
-            httpResponse.addHeader("Content-Disposition",
-                    "attachment; filename=\"" + request.getTableName() + ".csv\"");
-            httpResponse.addHeader("Content-Length", String.valueOf(bytes.length));
-            httpResponse.addHeader("Content-Type", "text/csv;charset=utf-8"); // the current impl is csv
+            int remaining = request.getLimit();
+            int totalRecords = 0;
+            httpResponse.addHeader("Content-Type", recordsExporter.getWebMediaType());
+            httpResponse.addHeader(
+                    "Content-Disposition",
+                    "attachment; filename=\"" + extractRealTablename(request.getTableName()) + "."
+                            + recordsExporter.getFileSuffix() + "\""
+            );
             ServletOutputStream outputStream = httpResponse.getOutputStream();
-            outputStream.write(bytes);
+            do {
+                RecordList recordList = queryRecordList(request);
+                int resultSize = recordList.getRecords().size();
+                if (resultSize == 0) {
+                    break;
+                }
+                recordsExporter.exportTo(recordList, outputStream);
+                totalRecords += resultSize;
+                request.setStart(totalRecords);
+                if (remaining != -1) {
+                    remaining = remaining - resultSize;
+                    if (remaining <= 0) {
+                        break;
+                    }
+                }
+                request.setLimit(remaining);
```

Comment:
Limit the `limit`?
https://github.com/star-whale/starwhale/blob/12e0fd2887f674e007715ace6a099ae380d53566/server/controller/src/main/java/ai/starwhale/mlops/datastore/DataStore.java#L190-L194

Reply:
done